app/controllers/user_controller.rb

```diff
@@ -381,7 +381,8 @@ def api_read
 
   def api_details
     @this_user = @user
-    render :action => :api_read, :content_type => "text/xml"
+    allow_read_email = !current_token.nil? && current_token.read_attribute(:allow_read_email)
```

Comment:
Isn't that equivalent to the (simpier) form `current_token && current_token.read_attribute(:allow_read_email)`?